### PR TITLE
[Project] Log skipped and up-to-date projects in counts

### LINKED ISSUE #5431: Log skipped and up-to-date projects in counts
If I build a solution in VS where one project is up to date and one is disabled, I get the following message:

```
1>------ Skipped Build: Project: NetStandardLib, Configuration: Debug Any CPU ------
1>Project not selected to build for this solution configuration 
========== Build: 0 succeeded, 0 failed, 1 up-to-date, 1 skipped ==========
```

MD only logs the succeed/fail count, it does not log the number of skipped or up to date projects.

> VS bug [#650153](https://devdiv.visualstudio.com/DevDiv/_workitems/edit/650153)

### PR DESCRIPTION
**Pull request by mhutch**
Fixes #5431

### CODE REVIEW

File: main/src/core/MonoDevelop.Core/MonoDevelop.Projects/BuildResult.cs [hunk 1]
```diff
@@ -35,24 +35,22 @@ namespace MonoDevelop.Projects
...
 		List<BuildError> errors = new List<BuildError> ();
 		IBuildTarget sourceTarget;
 
 		public BuildResult()
 		{
+			BuildCount = 1;
 		}
 		
-		public BuildResult (string compilerOutput, int buildCount, int failedBuildCount)
+		public BuildResult (string compilerOutput, int buildCount, int failedBuildCount) : this ()
```
Comment (Therzok) [inline main/src/core/MonoDevelop.Core/MonoDevelop.Projects/BuildResult.cs]: Why call `this()` here?

**Review by Therzok — APPROVED** [review]

File: main/src/core/MonoDevelop.Core/MonoDevelop.Projects/BuildResult.cs (hunk 1 above)
Comment (mhutch) [reply main/src/core/MonoDevelop.Core/MonoDevelop.Projects/BuildResult.cs]: Consistency, but fair point, it's not necessary :)

Comment (mhutch) [reply main/src/core/MonoDevelop.Core/MonoDevelop.Projects/BuildResult.cs]: OK, cleaned things up a bit

**Review by Therzok — APPROVED** [review]

**Comment by mhutch** [comment]
Test failure is unrelated

**Review by slluis — APPROVED** [review]

**Pull request merged**